hack/scripts/prow-ci-operator-deploy.sh

```diff
@@ -1,4 +1,205 @@
+## Remove Below Line
+OC="oc"
+OC_WITH_NAMESPACE="$OC -n $NAMESPACE"
+
+function usage {
+    cat <<EOF
+    USAGE: $0 [ OPTION ]
+
+    OPTIONS:
+
+    --use-envrc     Uses the .envrc for setting up configmap and secret otherwise pick up from ci vault secrets
+    -n|--namespace  Sets the namespace to use
+    --is-local      Uses settings and makefile targets for local deployment
+    --is-staging    Uses settings and makefile targets for staging deployment
+EOF
+}
+
+function parseArgs {
+    PARSED_ARGUMENTS=$(getopt -o 'n:' --long 'namespace:,use-envrc,is-local,is-staging' -- "$@")
+    eval set -- "$PARSED_ARGUMENTS"
+
+    while :
+    do
+        case "$1" in
+            --use-envrc)
+                USE_ENVRC=1; shift
+                ;;
+            --is-local)
+                IS_LOCAL=1; shift
+                ;;
+            --is-staging)
+                IS_STAGING=1; shift
+                ;;
+            -n|--namespace)
+                NAMESPACE="$2";	shift 2
+                ;;
+            --)
+                shift
+                break
+                ;;
+            *)
+                echo "Unexpected option: $1."
+                usage
+                break
+                ;;
+        esac
+    done
+
+    echo "USE_ENVRC=${USE_ENVRC}"
+    echo "NAMESPACE=${NAMESPACE}"
+    echo "IS_LOCAL=${IS_LOCAL}"
+    echo "IS_STAGING=${IS_STAGING}"
+}
+
+function cleanup {
+    echo -e "\nPERFORMING CLEANUP\n"
+    ##  Add step to remove CRDs
+    if [[ $($OC get namespace $NAMESPACE --no-headers | wc -l) == 0 ]];
+        then
+            echo -e "\nno $NAMESPACE namespace found\n"
+            return
+	fi
+
+    $OC delete namespace $NAMESPACE || true
+}
+
+function cleanupPre {
+    cleanup    
+}
+
+function getEnvVariables {
+    if [[ -z $USE_ENVRC ]];
+        then
+            export FAKE_KEY_VALUE=$(cat /tmp/secret/aao-aws-creds/FAKE_KEY_VALUE)
+            export FORCE_DEV_MODE=cluster
+            export OSD_STAGING_1_AWS_ACCOUNT_ID=$(cat /tmp/secret/aao-aws-creds/OSD_STAGING_1_AWS_ACCOUNT_ID)
+            export OSD_STAGING_2_AWS_ACCOUNT_ID=$(cat /tmp/secret/aao-aws-creds/OSD_STAGING_2_AWS_ACCOUNT_ID)
+            export STS_ROLE_ARN=$(cat /tmp/secret/aao-aws-creds/STS_ROLE_ARN)
+            export STS_JUMP_ARN=$(cat /tmp/secret/aao-aws-creds/STS_JUMP_ARN)
+            export OSD_STAGING_1_OU_ROOT_ID=$(cat /tmp/secret/aao-aws-creds/OSD_STAGING_1_OU_ROOT_ID)
+            export OSD_STAGING_1_OU_BASE_ID=$(cat /tmp/secret/aao-aws-creds/OSD_STAGING_1_OU_BASE_ID)
+            export SUPPORT_JUMP_ROLE=$(cat /tmp/secret/aao-aws-creds/SUPPORT_JUMP_ROLE)
+            export STS_JUMP_ROLE=$(cat /tmp/secret/aao-aws-creds/STS_JUMP_ROLE)
+            export OPERATOR_ACCESS_KEY_ID=$(cat /tmp/secret/aao-aws-creds/OPERATOR_ACCESS_KEY_ID)
+            export OPERATOR_SECRET_ACCESS_KEY=$(cat /tmp/secret/aao-aws-creds/OPERATOR_SECRET_ACCESS_KEY)
+        else
+            source .envrc
```

Comment:
Might want to check the file actually exists, before attempting to source it and error out if it does not.

Reply:
Added a check for .envrc before usage.